Tests/NIOWebSocketTests/EndToEndTests.swift

```diff
@@ -397,7 +397,7 @@ class EndToEndTests: XCTestCase {
                          expectedResponseLine: "HTTP/1.1 101 Switching Protocols",
                          expectedResponseHeaders: ["Upgrade: websocket", "Sec-WebSocket-Accept: OfS0wDaT5NoxF2gqm7Zj2YtetzM=", "Connection: upgrade"])
 
-        let decoder = (try server.pipeline.context(handlerType: WebSocketFrameDecoder.self).wait()).handler as! WebSocketFrameDecoder
+        let decoder = ((try server.pipeline.context(handlerType: ByteToMessageHandler<WebSocketFrameDecoder>.self).wait()).handler as! ByteToMessageHandler<WebSocketFrameDecoder>).decoder
```

Comment:
Hrm. This is a bad test, we should fix it. It should test this by attempting to *violate* the max frame size rule and seeing that that fails.

Reply:
Appreciate that we probably shouldn't fix it as part of this PR though.